hivemq-edge/src/frontend/src/components/rjsf/BatchSubscription/hooks/useBatchModeSteps.tsx

```diff
@@ -0,0 +1,67 @@
+import { useSteps } from '@chakra-ui/react'
+import { useTranslation } from 'react-i18next'
+import { useCallback } from 'react'
+
+import DataSourceStep from '@/components/rjsf/BatchSubscription/components/DataSourceStep.tsx'
+import SubscriptionsValidationStep from '@/components/rjsf/BatchSubscription/components/SubscriptionsValidationStep.tsx'
+import ColumnMatcherStep from '@/components/rjsf/BatchSubscription/components/ColumnMatcherStep.tsx'
+import ConfirmStep from '@/components/rjsf/BatchSubscription/components/ConfirmStep.tsx'
+
+export enum BatchModeStep {
+  UPLOAD,
+  MATCH,
+  VALIDATE,
+  CONFIRM,
+}
+
+export interface BatchModeSteps {
+  id: BatchModeStep
+  title: string
+  description: string
+  renderer: JSX.Element
+  isFinal?: boolean
+}
+
+export const useBatchModeSteps = () => {
+  const { t } = useTranslation('components')
+  const { isCompleteStep, isIncompleteStep, ...stepper } = useSteps()
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const isStepCompleted = useCallback((_: BatchModeStep): boolean => {
+    return true
+  }, [])
```

Comment:
question: that one is a placeholder too, right?

otherwise it should be this :D

```suggestion
  const isStepCompleted = true
```